### PR TITLE
docs: Adds `Icon` component page

### PR DESCRIPTION
**Pull request by hellofanny**
## What's the purpose of this pull request?

- Adds missing Icon component [doc page](https://faststore-site-git-doc-icon-doc-page-fs-862-faststore.vercel.app/components/atoms/icon)

### CODE REVIEW

File: apps/site/pages/components/atoms/icon.mdx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+<header>
+
+# Icon
+
+Icons are simplified graphics used to represent concepts and help users navigate the UI.
+
+</header>
+
+## Overview
```
Comment (renatamottam) [inline apps/site/pages/components/atoms/icon.mdx:20]: I think we should mention `Phosphor Icons` here. Should we also explain how to add new icons?

File: apps/site/pages/components/atoms/icon.mdx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+<OverviewSection>
+  <Icon component={<ShoppingCart />} />
```
Comment (renatamottam) [inline apps/site/pages/components/atoms/icon.mdx:23]: Also, shouldn't we list all available icons here?

File: apps/site/pages/components/atoms/icon.mdx (hunk 1 above)
Comment (hellofanny) [reply apps/site/pages/components/atoms/icon.mdx:20]: Agreed! I’ll add it!

File: apps/site/pages/components/atoms/icon.mdx (hunk 2 above)
Comment (hellofanny) [reply apps/site/pages/components/atoms/icon.mdx:23]: I was thinking about it! We can list the icons that we are currently using for our components.

**Review by renatamottam — APPROVED** [review]

**Pull request merged**